feat: split software eng role into 3 roles (#338)
shorebirdtech/website: src/content/jobs/web-software-engineer.md

---
title: Web Software Engineer
description:
  Own our front-ends and work with designers to optimize user experience for
  Shorebird customers.
location:
  Remote (North America) or in-person (Chicago, New York City, or San Francisco)
type: Full-time
cover: engineer-cover.png
---

### Overview

We have one successful product with thousands of monthly users, delivering 10s
of millions of updates around the globe, every month. We built this all with 2
engineers. We also recently launched a second product which is also growing. We
simply need more engineers to keep up with our growing user base and ambition.

### What You'll Do

- Own our front-ends. This includes our website, web console, admin panel, and other projects.

- Work with other engineers to implement the front-end of new features and
  products.

- Build the user experience for all the cloud parts that a Shorebird customer
  needs, but doesn't want to build themselves (e.g. anything Firebase offers,
  various GitHub-adjacent services, etc.).

- Work with designers to optimize for user experience.

- Help shape our culture, processes, and product direction.

> - Note: We don't use Flutter, we build Flutter itself and services for Flutter.

Reviewer: Nit, but I'm not sure this line is necessary in this particular job req

- Located in North America, ideally Chicago, New York City or San Francisco.

- 2+ years of experience shipping production web apps.

- Experience with React, TypeScript, Postgres and (ideally) Dart (what our
  backend is written in).

- Nice to have: Startup experience, ideally B2B SaaS.

- You're a self-starter — you work well with little oversight and thrive in
  ambiguous environments. You'll have support as a member of our engineering
  team, but this still won't be like working at a big tech job with formal
  training or processes, you'll have to learn and teach yourself most things
  (with help).

- You care about product quality and developer experience.

- Bonus: startup experience or contributions to large open source projects.

### Why Join

- Work directly with a small, senior team solving hard, high-impact problems.

- Competitive compensation + early equity.

- Shape the future of how mobile apps are built and shipped.

- This job is a great opportunity for someone who wants to work at a startup,
  has done some frontend but is generally interested in the opportunities that
  come from working on such a small and senior team.

> - Access to 1000s of customers via our Discord if you need to ask one something.

Reviewer: ```suggestion
- Direct access to 1000s of customers via our Discord.
```
